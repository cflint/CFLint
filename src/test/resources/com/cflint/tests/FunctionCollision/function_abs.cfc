component {

public any function abs(){
    x=123;
  }
}