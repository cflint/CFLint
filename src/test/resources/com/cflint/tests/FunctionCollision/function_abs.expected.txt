{
  "version" : "",
  "timestamp" : 1568428224,
  "issues" : [ {
    "severity" : "WARNING",
    "id" : "FUNCTION_NAME_COLLISION",
    "message" : "FUNCTION_NAME_COLLISION",
    "category" : "CFLINT",
    "abbrev" : "FN",
    "locations" : [ {
      "file" : "src\\test\\resources\\com\\cflint\\tests\\FunctionCollision\\function_abs.cfc",
      "fileName" : "function_abs.cfc",
      "function" : "abs",
      "offset" : 33,
      "column" : 21,
      "line" : 3,
      "message" : "Avoid using the name abs for a function.  It is reserved in some CFML implementations.  See https://cfdocs.org/abs",
      "variable" : "abs",
      "expression" : "abs"
    } ]
  } ],
  "counts" : {
    "totalFiles" : 0,
    "totalLines" : 0,
    "countByCode" : [ {
      "code" : "FUNCTION_NAME_COLLISION",
      "count" : 1
    } ],
    "countBySeverity" : [ {
      "severity" : "WARNING",
      "count" : 1
    } ]
  }
}